tuftool/tests/create_repository_integration.rs

```diff
@@ -0,0 +1,268 @@
+            format!("{}", timestamp_version).as_str(),
+        ])
+        .assert()
+        .success();
+
+    // Load our newly created repo
+    let metadata_base_url = &test_utils::dir_url(repo_dir.path().join("metadata"));
+    let targets_base_url = &test_utils::dir_url(repo_dir.path().join("targets"));
+    let repo = Repository::load(
+        &tough::FilesystemTransport,
+        Settings {
+            root: File::open(root_json).unwrap(),
+            datastore: load_dir.as_ref(),
+            metadata_base_url,
+            targets_base_url,
+            limits: Limits::default(),
+            expiration_enforcement: ExpirationEnforcement::Safe,
+        },
+    )
+    .unwrap();
+
+    // Ensure we can read the targets
+    assert_eq!(
+        test_utils::read_to_end(repo.read_target("file1.txt").unwrap().unwrap()),
+        &b"This is an example target file."[..]
+    );
+    assert_eq!(
+        test_utils::read_to_end(repo.read_target("file2.txt").unwrap().unwrap()),
+        &b"This is an another example target file."[..]
+    );
+    assert_eq!(
+        test_utils::read_to_end(repo.read_target("file3.txt").unwrap().unwrap()),
+        &b"This is role1's target file."[..]
+    );
+
+    // Ensure the targets.json file is correct
+    assert_eq!(repo.targets().signed.version.get(), targets_version);
+    assert_eq!(repo.targets().signed.expires, targets_expiration);
+    assert_eq!(repo.targets().signed.targets.len(), 3);
+    assert_eq!(repo.targets().signed.targets["file1.txt"].length, 31);
+    assert_eq!(repo.targets().signed.targets["file2.txt"].length, 39);
+    assert_eq!(repo.targets().signed.targets["file3.txt"].length, 28);
+    assert_eq!(repo.targets().signatures.len(), 1);
+
+    // Ensure the snapshot.json file is correct
+    assert_eq!(repo.snapshot().signed.version.get(), snapshot_version);
+    assert_eq!(repo.snapshot().signed.expires, snapshot_expiration);
+    assert_eq!(repo.snapshot().signed.meta.len(), 1);
+    assert_eq!(
+        repo.snapshot().signed.meta["targets.json"].version.get(),
+        targets_version
+    );
+    assert_eq!(repo.snapshot().signatures.len(), 1);
+
+    // Ensure the timestamp.json file is correct
+    assert_eq!(repo.timestamp().signed.version.get(), timestamp_version);
+    assert_eq!(repo.timestamp().signed.expires, timestamp_expiration);
+    assert_eq!(repo.timestamp().signed.meta.len(), 1);
+    assert_eq!(
+        repo.timestamp().signed.meta["snapshot.json"].version.get(),
+        snapshot_version
+    );
+    assert_eq!(repo.snapshot().signatures.len(), 1);
+    root_json_dir.close().unwrap();
+}
+
+#[test]
+#[cfg_attr(not(feature = "integration-tests"), ignore)]
+// Ensure we can use local rsa key to create and sign a repo created by the `tuftool` binary using the `tough` library
+fn create_repository_local_key() {
+    let root_key_dir = TempDir::new().unwrap();
+    let root_key_path = root_key_dir.path().join("local_key.pem");
```

Comment:
I am creating a new one because I wanted to test end to end. I created a separate method for whole repository creation process, So that we can just use different types of key and ensure they all adhere to the same workflow of consumption. Unfortunately, `tough-kms` had to add an extra flag for creation, but it will be same in future and this test cases ensures that.

Reply:
I misread the test, I understand now.